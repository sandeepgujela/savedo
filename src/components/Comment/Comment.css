.comment-wrapper{
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
}

.comment-wrapper .user-image img{
    border-radius: 3px;
    width: 47px;
    /* display: none; */
    height: 47px;
    margin-right: 15px;
}

.comment-wrapper .comment-data{
    width:100% ;
    max-width: 768px;
    font-size: 14px;
    border:1px solid #e0e4e8;
    border-radius:3px;
    margin-bottom:20px;
}

.comment-wrapper .user-name{
    color: #808182;
    font-size: 15px;
    background-color: #f6f8fa;
    padding:15px;
    border-bottom: 1px solid #e0e4e8;
}

.comment-wrapper .body{
    padding:15px;
    word-break: break-all;
}


@media only screen and (max-width: 620px) {
.comment-wrapper .user-image img{
    display: none;
}
}
