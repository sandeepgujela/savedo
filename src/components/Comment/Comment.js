import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Comment.css';
import ReactMarkdown from 'react-markdown';
import { meaningFullTime } from '../../util/helpers';

class Comment extends Component {
  render() {
    const {
      user: { avatar_url, login },
      created_at,
      body,
    } = this.props.data;
    return (
      <div className="comment-wrapper">
        <div className="user-image">
          {avatar_url && <img src={avatar_url} alt="user avatar" />}
        </div>

        <div className="comment-data">
          <div className="user-name">
            <b>{login} </b>
            commented On {meaningFullTime(created_at)}
          </div>
          <div className="body">
            <ReactMarkdown source={body} />
          </div>
        </div>
      </div>
    );
  }
}

Comment.propTypes = {
  data: PropTypes.shape({
    user: PropTypes.shape({
      login: PropTypes.string.isRequired,
      avatar_url: PropTypes.string.isRequired,
    }).isRequired,
    created_at: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
  }).isRequired,
};

export default Comment;
