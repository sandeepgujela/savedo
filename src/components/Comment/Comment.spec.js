import React from 'react';
import Comment from './Comment';
import { shallow } from 'enzyme';

describe('<Comment />', () => {
  it('Should render comment component', () => {
    const props = {
      user: {
        avatar_url: 'some_url',
        login: 'savedo',
      },
      created_at: 'xx.yy.zzzz',
      body: '123 1233',
    };
    const component = shallow(<Comment data={props} />);
    expect(component).toMatchSnapshot();
  });
});
