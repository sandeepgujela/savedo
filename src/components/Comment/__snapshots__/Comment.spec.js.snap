// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> Should render comment component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comment
    data={
      Object {
        "body": "123 1233",
        "created_at": "xx.yy.zzzz",
        "user": Object {
          "avatar_url": "some_url",
          "login": "savedo",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-image"
        >
          <img
            alt="user avatar"
            src="some_url"
          />
        </div>,
        <div
          className="comment-data"
        >
          <div
            className="user-name"
          >
            <b>
              savedo
               
            </b>
            commented On 
            just now
          </div>
          <div
            className="body"
          >
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={true}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              source="123 1233"
              sourcePos={false}
              transformLinkUri={[Function]}
            />
          </div>
        </div>,
      ],
      "className": "comment-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="user avatar"
            src="some_url"
          />,
          "className": "user-image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "user avatar",
            "src": "some_url",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="user-name"
            >
              <b>
                savedo
                 
              </b>
              commented On 
              just now
            </div>,
            <div
              className="body"
            >
              <ReactMarkdown
                astPlugins={Array []}
                escapeHtml={true}
                plugins={Array []}
                rawSourcePos={false}
                renderers={Object {}}
                skipHtml={false}
                source="123 1233"
                sourcePos={false}
                transformLinkUri={[Function]}
              />
            </div>,
          ],
          "className": "comment-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  savedo
                   
                </b>,
                "commented On ",
                "just now",
              ],
              "className": "user-name",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "savedo",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "savedo",
                  " ",
                ],
                "type": "b",
              },
              "commented On ",
              "just now",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ReactMarkdown
                astPlugins={Array []}
                escapeHtml={true}
                plugins={Array []}
                rawSourcePos={false}
                renderers={Object {}}
                skipHtml={false}
                source="123 1233"
                sourcePos={false}
                transformLinkUri={[Function]}
              />,
              "className": "body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "astPlugins": Array [],
                "escapeHtml": true,
                "plugins": Array [],
                "rawSourcePos": false,
                "renderers": Object {},
                "skipHtml": false,
                "source": "123 1233",
                "sourcePos": false,
                "transformLinkUri": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-image"
          >
            <img
              alt="user avatar"
              src="some_url"
            />
          </div>,
          <div
            className="comment-data"
          >
            <div
              className="user-name"
            >
              <b>
                savedo
                 
              </b>
              commented On 
              just now
            </div>
            <div
              className="body"
            >
              <ReactMarkdown
                astPlugins={Array []}
                escapeHtml={true}
                plugins={Array []}
                rawSourcePos={false}
                renderers={Object {}}
                skipHtml={false}
                source="123 1233"
                sourcePos={false}
                transformLinkUri={[Function]}
              />
            </div>
          </div>,
        ],
        "className": "comment-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="user avatar"
              src="some_url"
            />,
            "className": "user-image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "user avatar",
              "src": "some_url",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="user-name"
              >
                <b>
                  savedo
                   
                </b>
                commented On 
                just now
              </div>,
              <div
                className="body"
              >
                <ReactMarkdown
                  astPlugins={Array []}
                  escapeHtml={true}
                  plugins={Array []}
                  rawSourcePos={false}
                  renderers={Object {}}
                  skipHtml={false}
                  source="123 1233"
                  sourcePos={false}
                  transformLinkUri={[Function]}
                />
              </div>,
            ],
            "className": "comment-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    savedo
                     
                  </b>,
                  "commented On ",
                  "just now",
                ],
                "className": "user-name",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "savedo",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "savedo",
                    " ",
                  ],
                  "type": "b",
                },
                "commented On ",
                "just now",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ReactMarkdown
                  astPlugins={Array []}
                  escapeHtml={true}
                  plugins={Array []}
                  rawSourcePos={false}
                  renderers={Object {}}
                  skipHtml={false}
                  source="123 1233"
                  sourcePos={false}
                  transformLinkUri={[Function]}
                />,
                "className": "body",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "astPlugins": Array [],
                  "escapeHtml": true,
                  "plugins": Array [],
                  "rawSourcePos": false,
                  "renderers": Object {},
                  "skipHtml": false,
                  "source": "123 1233",
                  "sourcePos": false,
                  "transformLinkUri": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
